scripts/reana

```diff
@@ -0,0 +1,112 @@
+# submit itself to any jurisdiction.
+
+reana_components="reana-job-controller reana-workflow-controller
+                  reana-workflow-monitor reana-message-broker
+                  reana-workflow-engine-yadage"
+
+install_local_minikube () {
+    # Check whether the OS is supported
+    if ! echo "darwin linux" | grep -F -q -w -i "$(uname -s)"
+    then
+        echo 'Your OS is not supported.'
+        exit 1
+    else
+        OS="$(uname -s | tr '[:upper:]' '[:lower:]')"
+    fi
+
+    # Install tools `kubectl` and `minikube` if necessary
+    if ! command -v kubectl > /dev/null 2>&1
+    then
+        echo 'Installing latest kubectl ...'
+        curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/$OS/amd64/kubectl"
+        chmod +x ./kubectl
+        sudo mv ./kubectl /usr/local/bin/kubectl
+    fi
+
+    if ! command -v minikube > /dev/null 2>&1
+    then
+        echo 'Installing minikube ...'
+        curl -Lo minikube "https://storage.googleapis.com/minikube/releases/v0.17.1/minikube-$OS-amd64"
+        chmod +x minikube
+        sudo mv minikube /usr/local/bin/
+    fi
+
+    # FIXME find better way to define this dependency.
+    # `extras_require` does not support not PyPI deps and
+    # `dependency_links` does not seem to work.
+    pip install -r requirements-k8s.txt
+
+    # Start `minikube`
+    minikube start || echo 'Error while starting minikube' && exit 1
+
+    # Use minikube Docker daemon, so local images will be used instead
+    # of pulling from registry
+    echo 'Using minikube Docker daemon ...'
+    eval "$(minikube docker-env)"
+
+    # Pull all REANA components images and retag for local `dev`
+    echo 'Pulling REANA components images ...'
+    for component in $reana_components
+    do
+        docker pull "reanahub/$component"
+        docker tag "reanahub/$component" "$component"
+    done
+
+    # Generate Kubernetes manifests
+    echo 'Building REANA configuration manifests ...'
+    reana-resources-k8s build-manifests
+
+    # Set current cluster secret for `reana-job-controller` component
+    current_cluster_secret=$(kubectl get secrets | grep 'service-account-token' | cut -d ' ' -f 1)
+    sed -i'' -e "s/default-token-8p260/$current_cluster_secret/" configuration-manifests/deployments/job-controller.yaml
+
+    # Deploy system
+    echo 'Deploying REANA on minikube ...'
+    kubectl create -Rf configuration-manifests
+
+    # Clean configuration manifests
+    rm -rf configuration-manifests
+}
+
+usage() {
+    echo "${0##*/} manages REANA system operations"
+    echo ""
+    echo "Usage:"
+    echo "  install-minikube    Deploys local minikube based REANA cluster"
+    echo "  -h                  Help"
+}
+
+if [ "$1" != "" ]
+then
+    case $1 in
+        install-minikube ) install_local_minikube
+                           ;;
+        clean-minikube)    minikube delete
```

Comment:
I couldn't get minikube cluster to start after the first run without running ``` minikube delete ``` first.
I'm not sure how many times I started it before using ``` reana install-minikube ``` or did I mess around with Virtualbox, but I would bet that some of the minikube VM's were either in corrupted state or not present.
Maybe something about running ``` reana clean-minikube ``` should be added to the documentation? Maybe just a side note:

_"NOTE: In case you run into error messages from Kubernetes related to local Kubernetes cluster (aka minikube), you can try to run ``` reana clean-minikube ``` to delete VMs created by previous minikube installation / initialization"_

BTW, the error I got was:
``` $ reana install-minikube
Starting local Kubernetes cluster...
Starting VM...
E0403 20:06:00.759374   16086 start.go:113] Error starting host: Error getting state for host: machine does not exist.
```